api/v1alpha1/nodepool_types.go

```diff
@@ -317,6 +317,53 @@ type NodePoolPlatform struct {
 	Agent *AgentNodePoolPlatform `json:"agent,omitempty"`
 
 	Azure *AzureNodePoolPlatform `json:"azure,omitempty"`
+
+	// PowerVS specifies the configuration used when using IBMCloud PowerVS platform.
+	//
+	// +optional
+	PowerVS *PowerVSNodePoolPlatform `json:"powervs,omitempty"`
+}
+
+// PowerVSNodePoolPlatform specifies the configuration of a NodePool when operating
+// on IBMCloud PowerVS platform.
+type PowerVSNodePoolPlatform struct {
+	// SysType used to host the instance(e.g: s922, e980, e880)
+	// +optional
```

Comment:
for all the fields that have defaults can you pleasse include that in the comment so it's exposed in autogenerated docs. e.g:
```
// SysType used to host the instance(e.g: s922, e980, e880)
// The default is s922
```
For immutable fields:

```
// This field is immutable. Once set, It can't be changed.
```

Reply:
some of these defaults are influenced from the default node pool set by the installer here: https://github.com/openshift/installer/blob/c60a959e5278e6d2080f84d2fd31531390cc66c6/pkg/asset/machines/worker.go#L170, contrast to this having more required params is going to impact the UX is what I feel, this is like setting the default profile for the machine in other providers

Reply:
addressed the comment in the recent comment, ptal

Reply:
btw: `+immutable` is noop at the moment, ref: https://github.com/openshift/hypershift/commit/b0e61e5399e29a8df12c9463a939b634a7a56245 more information, not sure how this text is going to prevent user to do so 🤔